### PR TITLE
Rewriting GraphQL requests for constant extraction

### PR DESCRIPTION
**Pull request by tailhook**
There are two kinds of failing tests:
1. Variables in introspection queries aren't part of the cache key
2. Transforming errors on variables to errors on constants

*(Note: this PR is on top of `graphql_core_3` branch, so approvals are unuseful here, I'll recreate PR when that branch is landed)*

### CODE REVIEW

**Comment by ambv** [comment]
Man, that's a huge change.

**Comment by tailhook** [comment]
> Man, that's a huge change.

Well, generally it's because it includes the whole graphql-parser and its tests. Perhaps, you can compare starting from that: https://github.com/edgedb/edgedb/compare/f135c5ca7d9e49af8825798691f56f6b912e8fa1...graphql_rewrite . Pretty small: 
image

File: tests/test_http_graphql_query.py [hunk 1]
```diff
@@ -1190,8 +1190,12 @@ def test_graphql_functional_arguments_21(self):
     def test_graphql_functional_arguments_22(self):
         with self.assertRaisesRegex(
                 edgedb.QueryError,
-                r"invalid value for 'after'",
-                _line=5, _col=32):
+                # this error message is subpar, but this is what we get
+                # from postgres, because we transfer bigint values to postgres
+                # as strings
+                r'invalid input syntax for type bigint: "aaaaa"',
```
Comment (vpetrovykh) [inline tests/test_http_graphql_query.py]: There's a historical reason why this is a string and the short version of the story is that this was inspires by Relay Connections (see here: https://github.com/edgedb/edgedb/issues/187). Depending on whether we still find that style useful here, maybe we can change the type to be an integer like "first" and "last".

**Comment by vpetrovykh** [comment]
Other than my non-blocking comment, this looks fine.

File: tests/test_http_graphql_query.py (hunk 1 above)
Comment (tailhook) [reply tests/test_http_graphql_query.py]: Well, generally I don't think that pagination using `OFFSET` (which `after` and `before` are currently mapped to) is a good solution. Pagination should be coupled with sort order and use sorting key as `after` and `before`. But this is irrelevant to this PR. Please, create a new issue if you think this is worth changing now.

**Pull request closed without merging**